test/IDE/complete_diagnostics_concurrency.swift

```diff
@@ -30,3 +30,11 @@ func testActor(obj: MyActor) async {
 // ACTOR-DAG: Decl[InstanceMethod]/CurrNominal/NotRecommended: receiveNonSendable({#arg: MyNonSendable#})[' async'][#Void#]; name=receiveNonSendable(arg:); diagnostics=warning:actor-isolated 'receiveNonSendable(arg:)' should only be referenced from inside the actor{{$}}
 // ACTOR: End completions
 }
+
```

Comment:
Would be good to test the contextual and context-free cases.

Reply:
There's no case where `ContextFreeNotRecReason != None && AssociatedDecl == nulptr` at this point. Added an assertion to clarify, but still use `dyn_cast_or_null ` just in case.

Reply:
Seems reasonable, no crashes in release but we'll see with assertions enabled 👍